### DISCUSSION TITLE
如何引入这个库

### DISCUSSION THREAD

**Discussion by ozline** [west2-online/jwch#1] [Announcements]
由于这个库涉及到安全问题，因此这个库是Private的，我们需要进行一些设定来使用这个库

# 创建Token
1.  点击右上角头像->Settings
2. 点击左边列表中的Developer Settings->Personal access tokens
3. 点击Generate new token
4. Token name和Expire time可以随便填，建议过期时间长一点，比如一年
5. Resource Owner 选择**west2-online**
6. Repository access选择All Repository
7. Repository permissions中，Metadata设置为Read-Only，Contents设置为Read and Write
8. 点击底部的Generate token

# 应用Token
我们可以参考这篇文章：[Why does "go get" use HTTPS when cloning a repository?](https://go.dev/doc/faq#git_https)

直接复制创建好的token，最好暂时不要关闭这个页面，因为关闭后这个token就消失了

我们在`$HOME/.netrc`中添加如下内容
```
machine github.com login USERNAME password APIKEY
```
其中，USERNAME是你的github登录账号，APIKEY是刚才的token

保存好后，我们需要设置go环境 ，这里请先确保golang版本在1.13以上

我们使用如下语句设定
```
go env -w GOPRIVATE=github.com/west2-online
```
可以使用go env查看是否设定好

一切就绪后，我们可以在项目中使用
```
go get github.com/west2-online/jwch
```
来引入包了

**Comment by SkyDDDog** [comment]
> 我们在`$HOME/.netrc`中添加如下内容
> 
> ```
> machine github.com login USERNAME password APIKEY
> ```
> 
> 其中，USERNAME是你的github登录账号，APIKEY是刚才的token
> 

Windows用户添加.netrc文件参考下文:
* 定义一个用户环境变量%HOME%， 值为%USERPROFILE%
* 在%HOME%路径下新建一个文件_netrc
* 在_netrc文件中增加下面的配置
~~~text
machine github.com
login USERNAME 
password APIKEY
~~~

**Comment by ozline** [comment]
在今天对自己Repo进行上传的时候，我发现我需要删除.netrc文件才不会导致权限错误，而在删除后我可以正常的对这个库进行clone/push以及go get，那么这个.netrc似乎是不需要的，那为什么官方教程里说要这么弄呢..
